aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -571,48 +566,21 @@ Tensor& _sparse_csr_mm_out(
 }
 
 Tensor _sparse_csr_mm(const Tensor& mat1, const Tensor& mat2) {
-  if (mat1.is_sparse_csr() && mat2.is_sparse_csr()) {
```

Comment:
The layout checking here is only used to determine how result it initialized and is over complicated.